src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -397,6 +397,10 @@ private function darwinRealpath($real)
         $kFile = strtolower($file);
 
         if (!isset($dirFiles[$kFile])) {
+            if (') : eval()\'d code' === substr($kFile, -17)) {
```

Comment:
to allow the result to be cached, this should be before of the previous "if", using
`if (!isset($dirFiles[$kFile]) && ') : eval()\'d code' === substr($kFile, -17)) {` for example

Reply:
Done. I moved it before the first `isset()` as it might even save the `strtolower()` call.